first-step-spring-class-config/src/main/java/org/elearning/portfolio/AppConfigBean.java

```diff
@@ -0,0 +1,38 @@
+package org.elearning.portfolio;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.jdbc.datasource.DriverManagerDataSource;
+import javax.sql.DataSource;
+import org.elearning.portfolio.user.UserDAOImpl;
+import org.elearning.portfolio.services.UserService;
+import org.elearning.portfolio.message.MessageDAOImpl;
+import org.elearning.portfolio.services.MessageService;
+
+
+@Configuration
+public class AppConfigBean{
+
+    @Bean
+    public CreateDataSourceForJdbcTemplate createDataSourceForJdbcTemplate(){
+        return new CreateDataSourceForJdbcTemplate();
+    }
+    @Bean
+    public UserDAOImpl userDAOImpl(){
```

Comment:
Tak, to jest moment w którym stosujesz new i implementację danego interfejsu - jednak tutaj jako typ zwracany ustaw nadal interfejs a nie implementację - czyli zamiast UserDAOImpl ustaw UserDAO, podobnie w pozostałych miejscach w tej klasie